packages/nvidia_nat_langchain/src/nat/plugins/langchain/langgraph_workflow.py

```
@@ -107,15 +107,28 @@ def _convert_input(self, value: Any) -> LanggraphWrapperInput:
         try:
-            output = await self._graph.ainvoke(value.model_dump())
+            # Check if the graph is an async context manager (e.g., from @asynccontextmanager)
+            if hasattr(self._graph, '__aenter__') and hasattr(self._graph, '__aexit__'):
+                logger.info("Graph is an async context manager")
+                async with self._graph as graph:
+                    output = await graph.ainvoke(value.model_dump())
+            else:
+                output = await self._graph.ainvoke(value.model_dump())
             return LanggraphWrapperOutput.model_validate(output)
         except Exception as e:
             raise RuntimeError(f"Error in LangGraph workflow: {e}") from e
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)